src/main/java/com/splunk/kafka/connect/SplunkSinkConnectorConfig.java

```diff
@@ -497,4 +521,28 @@ private void validateTopicsAndTopicsRegexCombination(Map<String, String> taskCon
         }
     }
 
-}
+    private void validateRegexForTimestamp(String regex){
+            if(enableTimestampExtraction &&  regex == null ){
+                throw new ConfigException("For Timestamp Extraction regex value should not be null");
+            }
+            
+            if(StringUtils.isNotBlank(regex) && enableTimestampExtraction){
+                if(Pattern.compile(regex) == null){
+                    throw new ConfigException(" Invalid Regex please enter again");
+                }
+                if(!getNamedGroupCandidates(regex)){
+                    throw new ConfigException("Invalid Named Group");
+                }
+            }
+    }
+
+    private static boolean getNamedGroupCandidates(String regex) {
+        Matcher m = Pattern.compile("\\(\\?<([a-zA-Z][a-zA-Z0-9]*)>").matcher(regex);
+            while (m.find()) {
+                if(m.group(1).equals("time")) {
+                    return true;   
+                }             
+            }
+            return false;
+        }
```

Comment:
This one seems to be formatted strangely.

Reply:
Does Java have some code formatter that we can use to format all the code?

Reply:
Updated formatting.